<launch>
    <!-- Load camera parameters -->
    <rosparam command="load" file="$(find rgbd_camera_interface)/config/camera_params.yaml" />
    
    <node name="rgbd_camera" pkg="rgbd_camera_interface" type="rgbd_camera_interface_node" output="screen">
        <param name="left_camera_id" value="0"/>
        <param name="right_camera_id" value="1"/>
        <param name="frame_id" value="camera_link"/>
    </node>

    <!-- Optional: Add camera calibration node -->
    <node name="camera_calibration" pkg="camera_calibration" type="cameracalibrator.py" 
          args="--size 8x6 --square 0.108"
          output="screen"
          if="$(arg calibration_mode)">
        <remap from="left" to="/rgbd_camera/left/image_raw"/>
        <remap from="right" to="/rgbd_camera/right/image_raw"/>
    </node>

    <!-- Optional: Add image view nodes for debugging -->
    <node name="left_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/rgbd_camera/left/image_raw"/>
    </node>
    
    <node name="right_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/rgbd_camera/right/image_raw"/>
    </node>
    
    <node name="depth_view" pkg="image_view" type="image_view" respawn="false" output="screen">
        <remap from="image" to="/rgbd_camera/depth/image_raw"/>
    </node>
</launch>
